pkg/operator/ceph/controller/predicate.go

```diff
@@ -104,7 +104,8 @@ func WatchControllerPredicate[T client.Object](scheme *runtime.Scheme) predicate
 				logger.Debugf("resource %q: %q is going to be deleted", kind, nsName)
 				return true
 			} else if objOld.GetGeneration() != objNew.GetGeneration() {
-				logger.Debugf("skipping resource %q: %q, update event with unchanged spec", kind, nsName)
+				logger.Debugf("reconciling %s %q with changed generation", kind, nsName.String())
+				return true
```

Comment:
there is other instances of it below with this switch cases, I think we need to update all of them

Reply:
This should cover all the Rook CR updates, I don't see where else it's needed. Which cases are you referring to?

Reply:
Okay I was on the old branch